code/Model/SiteTree.php

```diff
@@ -471,16 +471,16 @@ public static function get_by_link($link, $cache = true)
         // Grab the initial root level page to traverse down from.
         $URLSegment = array_shift($parts);
         $conditions = [$urlSegmentExpr => rawurlencode($URLSegment ?? '')];
-        if (self::config()->get('nested_urls')) {
+        if (static::config()->get('nested_urls')) {
```

Comment:
This (and any other invocations of this particular config property) really should be `SiteTree::config()` - this config property seems very much intended to have a single value that affects the system as a whole, rather than resulting in different behaviours for different subclasses based on config for each subclass.

That said, it's possible that this could be (ab)used right now to suit project-specific use cases - so changing this to `SiteTree` now might cause bugs.

@emteknetnz Do you have any thoughts on this?